### ISSUE TITLE
python-unidiff does not work on Python 3.x

### ISSUE THREAD

**Issue by vmalloc** [matiasb/python-unidiff#26]
Tried: 
```python
>>> diff = requests.get('https://github.com/matiasb/python-unidiff/pull/3.diff').content.decode('utf-8')
>>> p = PatchSet(diff)
>>> p
<PatchSet: []>
```

**Comment by matiasb** [comment]
Yeah, I agree. Thanks for reporting the issue.

I guess you already worked-around the issue, but FTR until this gets fixed, PatchSet expects a file-like object (so you could use StringIO to build a file-like object from the string).

**Comment by MaxBittker** [comment]
```
In [27]: pkg_resources.get_distribution("unidiff").version
Out[27]: '0.5.4'
In [28]: diff = requests.get('https://github.com/matiasb/python-unidiff/pull/3.diff').content.decode('utf-8')
In [29]: p = PatchSet(diff)
In [30]: p
Out[30]: <PatchSet: []>
```
This still isn't working for me on the new version, but piping from curl into the CLI version does. Any advice?

**Comment by MaxBittker** [comment]
update: `PatchSet.from_string(diff_data)` does work!

**Comment by MaxBittker** [comment]
(I see now this was the intended behavior, but it is confusing) Would you accept a patch that infers the type in the constructor and acts accordingly?

**Comment by matiasb** [comment]
Right, PatchSet still expects a file-like object, but there is a new constructor, from_string:
```
>>> diff = requests.get('https://github.com/matiasb/python-unidiff/pull/3.diff').content.decode('utf-8')
>>> p = PatchSet.from_string(diff)
>>> p
<PatchSet: [<PatchedFile: .gitignore>, <PatchedFile: unidiff/patch.py>, <PatchedFile: unidiff/utils.py>]>
```
But it's true that it could fail if there is no file, instead of returning an empty PatchSet.

A patch would be welcome! Just in case, it should keep backwards compatibility.

**Comment by MaxBittker** [comment]
can probably close this, may want to document the new usage

**Comment by earonesty** [comment]
i generally prefer explicit factories (from_string, from_bytes) over inferred behavior.  things like multiple inheritance, builtins, isinstance can be tricky/break things when automagically deducing arguments
